src/Microsoft.Kusto.ServiceLayer/DataSource/KustoResultsReader.cs

```diff
@@ -11,26 +11,31 @@ internal class KustoResultsReader : DataReaderWrapper
         /// Kusto returns atleast 4 results tables - QueryResults(sometimes more than one), QueryProperties, QueryStatus and Query Results Metadata Table. 
         /// ADS just needs query results. When returning query results we need to trim off the last 3 tables. 
         /// </summary> 
-        public KustoResultsReader(IDataReader reader)
+        public KustoResultsReader(List <IDataReader> readers)
             : base()
         {
             // Read out all tables
             List<DataTable> results = new List<DataTable>();
-            while (!(reader?.IsClosed ?? true))
+
+            foreach (var reader in readers)
             {
-                DataTable dt = new DataTable();
-                dt.Load(reader); // This calls NextResult on the reader
-                results.Add(dt);
+                while (!(reader?.IsClosed ?? true))
+                {
+                    DataTable dt = new DataTable();
+                    dt.Load(reader); // This calls NextResult on the reader
+                    results.Add(dt);
+                }
+                
+                // Trim results
+                if(results.Count > 3) results.RemoveRange(results.Count - 3, 3);
             }
 
-            // Trim results
-            if(results.Count > 3) results.RemoveRange(results.Count - 3, 3);
-
             // Create a DataReader for the trimmed set
             _resultDataSet = new DataSet();
-            foreach(var result in results)
+            for(int i = 0; i < results.Count; i++)
             {
-                _resultDataSet.Tables.Add(result);
+                results[i].TableName = "Table_" + i;
```

Comment:
Not sure how this Table name shows up in the UX. Please check on this and do something which is on par with the SQL experience.

Reply:
This name doesn't show anywhere in UX, its just for identifying it with unique name in_resultDataSet since all the separate queries return tables with name Table_0